src/core/model/table/column_layout_relation_data.cpp

```diff
@@ -20,56 +20,45 @@ std::vector<int> ColumnLayoutRelationData::GetTuple(int tuple_index) const {
 }
 
 std::unique_ptr<ColumnLayoutRelationData> ColumnLayoutRelationData::CreateFrom(
-        model::IDatasetStream& data_stream, bool is_null_eq_null, int max_cols, long max_rows) {
+        model::IDatasetStream& data_stream, bool is_null_eq_null) {
     auto schema = std::make_unique<RelationalSchema>(data_stream.GetRelationName());
     std::unordered_map<std::string, int> value_dictionary;
     int next_value_id = 1;
-    const int null_value_id = -1;
-    int num_columns = data_stream.GetNumberOfColumns();
-    if (max_cols > 0) num_columns = std::min(num_columns, max_cols);
+    const int null_value_id = kNullValueId;
+    const size_t num_columns = data_stream.GetNumberOfColumns();
     std::vector<std::vector<int>> column_vectors = std::vector<std::vector<int>>(num_columns);
-    int row_num = 0;
     std::vector<std::string> row;
 
     while (data_stream.HasNextRow()) {
         row = data_stream.GetNextRow();
 
-        if (row.empty() && num_columns == 1) {
-            row.emplace_back("");
-        } else if ((int)row.size() != num_columns) {
-            LOG(WARNING) << "Skipping incomplete rows";
+        if (row.size() != num_columns) {
+            LOG(WARNING) << "Unexpected number of columns for a row, skipping (expected "
+                         << num_columns << ", got " << row.size() << ")";
```

Comment:
I think the warning message should also contain the previous message, information that such rows are skipped is important

Reply:
It already says "skipping", is it not enough?

Reply:
oops, didn't see it